test/company_schema.js

```diff
@@ -16,4 +18,41 @@ describe("# Company Schema tests", () => {
             CompanySchemaTester.maxLength("name", CompanyConstants.companyName.max_length);
         });
     });
+
+    describe("Hook tests", () => {
+        let company;
+
+        beforeAll(async () => {
+            company = await Company.create({
+                name: "first name"
+            });
+            const offer = {
+                title: "Test Offer",
+                publishDate: new Date(Date.now() - (DAY_TO_MS)),
+                publishEndDate: new Date(Date.now() + (DAY_TO_MS)),
+                description: "For Testing Purposes",
+                contacts: ["geral@niaefeup.pt", "229417766"],
+                jobType: "SUMMER INTERNSHIP",
+                fields: ["DEVOPS", "MACHINE LEARNING", "OTHER"],
+                technologies: ["React", "CSS"],
+                location: "Testing Street, Test City, 123",
+                owner: company._id,
+                ownerName: company.name
+            };
+
+            await Offer.create([offer, offer]);
+        });
+
+        afterAll(async () => {
+            await Offer.deleteMany({});
+            await Company.deleteMany({});
+        });
+        test("Should update offers ownerName on company name update", async () => {
+            await Company.findByIdAndUpdate(company._id, { name: "new name" }, { new: true });
+
+            const offers = await Offer.find({ owner: company._id });
+
+            expect(offers.every(({ ownerName }) => ownerName === "new name")).toBe(true);
+        });
```

Comment:
I think we should add an assertion here to check that before the update the offers' ownerName is not the "new name". Since if the Offers always have the "new name" this passes.